### PR TITLE
feat: support for make and n8n integration states

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Adds Integration States visibility for `make` and `n8n`:
image

We now need to pass this as a request body parameter in the make and n8n apps as well!

## Type of change
- [x] Enhancement (small improvements)

## How should this be tested?
- Use the create webhook API endpoint and pass in a "source" key in the json body with value as any of the allowed sources: "make", "n8n", "zapier", "user".
- Now once this webhook is created for the given source, it should reflect the same in the connection state.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/integrations/page.tsx [hunk 1]
```diff
@@ -14,12 +14,15 @@ import { getIntegrations } from "@formbricks/lib/integration/service";
...
-    getCountOfWebhooksBasedOnSource(environmentId, "user"),
-    getCountOfWebhooksBasedOnSource(environmentId, "zapier"),
-  ]);
+  const [environment, integrations, userWebhooks, zapierWebhooks, makeWebhooks, n8nwebhooks] =
+    await Promise.all([
+      getEnvironment(environmentId),
+      getIntegrations(environmentId),
+      getCountOfWebhooksBasedOnSource(environmentId, "user"),
+      getCountOfWebhooksBasedOnSource(environmentId, "zapier"),
+      getCountOfWebhooksBasedOnSource(environmentId, "make"),
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/integrations/page.tsx:23]: please rename function to `getWebhookCountBySource` to follow the current naming pattern better

File: apps/web/app/(app)/environments/[environmentId]/integrations/page.tsx [hunk 2]
```diff
@@ -14,12 +14,15 @@ import { getIntegrations } from "@formbricks/lib/integration/service";
 export default async function IntegrationsPage({ params }) {
   const environmentId = params.environmentId;
 
-  const [environment, integrations, userWebhooks, zapierWebhooks] = await Promise.all([
-    getEnvironment(environmentId),
-    getIntegrations(environmentId),
-    getCountOfWebhooksBasedOnSource(environmentId, "user"),
-    getCountOfWebhooksBasedOnSource(environmentId, "zapier"),
-  ]);
+  const [environment, integrations, userWebhooks, zapierWebhooks, makeWebhooks, n8nwebhooks] =
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/integrations/page.tsx:17]: the variable are named `zapierWebhooks`, `userWebhooks`, even though it's not the webhook but the count. Please rename to `zapierWebhookCount`, etc.

File: packages/database/schema.prisma [hunk 3]
```diff
@@ -34,6 +34,8 @@ enum PipelineTriggers {
 enum WehbhookSource {
```
Comment (mattinannt) [inline packages/database/schema.prisma:34]: the typo is unfortunate ;-) 
Can we also rename this with the migration? ;-)

File: packages/types/webhooks.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const ZWebhook = z.object({
   createdAt: z.date(),
   updatedAt: z.date(),
   url: z.string().url(),
-  source: z.enum(["user", "zapier"]),
+  source: z.enum(["user", "zapier", "make", "n8n"]),
```
Comment (mattinannt) [inline packages/types/webhooks.ts:10]: since we have the source twice, please introduce a new variable `ZWebhookSource`

**Review by mattinannt — CHANGES_REQUESTED** [review]

**Comment by ShubhamPalriwala** [comment]
Hey @mattinannt, I've made the requested changes 🚀 One thing to keep in mind is, the migration is not backward compatible ie the older states of the webhook source will not be stored since its creating a new column and deleting the old one

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala thanks a lot for the changes :-)💪🙏

**Pull request merged**